### PR TITLE
Don't copy directories

### PR DESCRIPTION
**Pull request by kevva**
Somewhat fixes an issue in https://github.com/sindresorhus/cpy/pull/4 where directories are created as files.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,23 +35,28 @@ module.exports = function (src, dest, opts, cb) {
...
-						return;
-					}
-
+			fs.lstat(src, function (err, stats) {
+				if (err) {
+					done(err);
+					return;
+				}
+
+				if (stats.isDirectory()) {
```
Comment (schnittstabil) [inline index.js]: Checking if a file is not a directory before writing to it is vulnerable to race conditions: another process may remove the file and create a directory between the calls to stats.isDirectory() and fs.createReadStream(src)…

Comment (kevva) [inline index.js]: How would you do it then?

Comment (schnittstabil) [inline index.js]: What about try-catch and checking on EISDIR?
But why is this a problem at all? Why shouldn't handle the caller these errors?

Comment (kevva) [inline index.js]: It does not error if it's a directory. It just creates a file.

Comment (schnittstabil) [inline index.js]: Strange, but if `src` is an directory, then `fs.createReadStream` and `fs.readSync` definitely throw `EISDIR` Errors. (node v0.12.0)

Comment (schnittstabil) [inline index.js]: You may also run your test against HEAD~1, to see this.

File: test.js [hunk 2]
```diff
@@ -78,6 +78,14 @@ describe('cpFile()', function () {
 			cb();
 		});
 	});
+
+	it('should not copy directories', function (cb) {
+		cpFile('node_modules', 'tmp', function (err) {
+			assert(!err, err);
+			assert(!fs.existsSync('tmp'));
```
Comment (schnittstabil) [inline test.js]: It's strange but `fs.existsSync` [will be deprecated](http://nodejs.org/api/fs.html#fs_fs_existssync_path).
Personally, I would use:
```
assert.throws(fs.statSync.bind(fs, 'tmp'), /ENOENT/);
```

File: test.js [hunk 3]
```diff
@@ -117,4 +125,9 @@ describe('cpFile.sync()', function () {
 		assertDateEqual(licenseStats.atime, tmpStats.atime);
 		assertDateEqual(licenseStats.mtime, tmpStats.mtime);
 	});
+
+	it('should not copy directories', function () {
+		cpFile.sync('node_modules', 'tmp');
+		assert(!fs.existsSync('tmp'));
```
Comment (schnittstabil) [inline test.js]: Same as above.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: @schnittstabil Is right about race condition with stat. Basically you should never use the result of fs.stat/exists for any following fs read/write operation as the state of truth might change between the calls.

**Comment by schnittstabil** [comment]
As far as I can see, the Errors are _thrown_ (see: [cpy cli fails silently](https://github.com/sindresorhus/cpy/issues/5)).

But the problem remains, and there are two issues to consider:
1. if an (arbitrary) error occurs, should `cp-file` remove an incomplete file?
2. if src is a directory, should the file creation be prevented or should this be handled by 1.?

**Comment by sindresorhus** [comment]
1. Yes
2. Prevented with a descriptive error message

**Comment by sindresorhus** [comment]
ping @kevva :)

**Comment by kevva** [comment]
I'll fix this later tonight.

**Comment by schnittstabil** [comment]
@kevva I've already done something (#5) May you take a look at it?

**Pull request closed without merging**